backend/controllers/userController.js

```diff
@@ -0,0 +1,61 @@
+const asyncHandler = require("express-async-handler");
+const generateToken = require("../utils/jsonUtils");
+const User = require("../models/userModel");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken")
+
+const registerUser = asyncHandler(async (req, res) => {
+  console.log("HERE");
+  const { name, email, password } = req.body;
+
+  const userExists = await User.findOne({ email });
+
+  if (userExists) {
+    res.status(400); // bad request
```

Comment:
Minor comment: I would suggest returning the status code 403 or 409 although keeping it with 400 is fine as well. (There's quite a few different opinions on this)

Reply:
I will keep it to 400 for now